src/domain/wallet-invoices/wallet-invoice-receiver.ts

```diff
@@ -0,0 +1,50 @@
+import { WalletCurrency, ZERO_CENTS, ZERO_SATS } from "@domain/shared"
+
+export const WalletInvoiceReceiver = async ({
+  walletInvoice,
+  receivedBtc,
+  usdFromBtc,
+  usdFromBtcMidPrice,
+}: WalletInvoiceReceiverArgs): Promise<
+  WalletInvoiceReceiver | DealerPriceServiceError | ValidationError
+> => {
+  const zeroBankFee = {
+    usdBankFee: ZERO_CENTS,
+    btcBankFee: ZERO_SATS,
+  }
+  const btcToCreditReceiver = receivedBtc
+
+  if (walletInvoice.recipientWalletDescriptor.currency === WalletCurrency.Btc) {
+    const usdToCreditReceiver = await usdFromBtcMidPrice(btcToCreditReceiver)
```

Comment:
the display currency logic should not be tied to usd

Reply:
I believe this was resolved in one of the recent refactors. It comes from [this method](https://github.com/GaloyMoney/galoy/pull/1241/files#diff-c55ec68c8d8791847ea8a4305f0229424d7f03a12af0911adaa3bf40c0ac9741R97) now.

Reply:
https://github.com/GaloyMoney/galoy/pull/1241#discussion_r905641710